.github/workflows/preview.yml

```diff
@@ -0,0 +1,91 @@
+name: preview
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - ci-preview
+
+env:
+  CARGO_TERM_COLOR: always
+  RUST_VERSION: 1.80.0
```

Comment:
Wondering if we still need that, since we have the `rust-toolchain.toml` already in the repo. If can be removed (I think in `ci.yml` it should be the same), easier to maintain.